### PR TITLE
Update CircleCI docs for current Cloud/v2/v3 configuration

### PR DESCRIPTION
**Pull request by epragtbeamtree**
Documentation might have changed, but correct setup is as above, as per https://circleci.com/docs/executor-intro#linux-vm

### CODE REVIEW

File: docs/supported_docker_environment/continuous_integration/circle_ci.md [hunk 1]
```diff
@@ -6,12 +6,13 @@ executor type in your `.circleci/config.yml` to be `machine` instead of the defa
...
 ```yml
-# Check https://circleci.com/docs/2.0/language-java/ for more details
-#
-version: 2
-machine: true
+
 jobs:
   build:
+    # Check https://circleci.com/docs/executor-intro#linux-vm for more details
+    #
```
Comment (kiview) [inline docs/supported_docker_environment/continuous_integration/circle_ci.md:13]:
```suggestion
```

File: docs/supported_docker_environment/continuous_integration/circle_ci.md [hunk 2]
```diff
@@ -6,12 +6,13 @@ executor type in your `.circleci/config.yml` to be `machine` instead of the defa
...
+    machine: true
+
```
Comment (kiview) [inline docs/supported_docker_environment/continuous_integration/circle_ci.md:15]:
```suggestion
```

File: docs/supported_docker_environment/continuous_integration/circle_ci.md [hunk 3]
```diff
@@ -6,12 +6,13 @@ executor type in your `.circleci/config.yml` to be `machine` instead of the defa
 Here is a sample CircleCI configuration that does a checkout of a project and runs maven:
 
 ```yml
-# Check https://circleci.com/docs/2.0/language-java/ for more details
-#
-version: 2
-machine: true
+
```
Comment (kiview) [inline docs/supported_docker_environment/continuous_integration/circle_ci.md:9]:
```suggestion
```

**Review by kiview — COMMENTED** [review]
This docs would be applicable for Circle-CI Cloud, v2 and v3, correct?

File: docs/supported_docker_environment/continuous_integration/circle_ci.md (hunk 1 above)
Comment (epragtbeamtree) [reply docs/supported_docker_environment/continuous_integration/circle_ci.md:13]: I believe so, but I've only tested it for Cloud.

File: docs/supported_docker_environment/continuous_integration/circle_ci.md [hunk 4]
```diff
@@ -6,12 +6,10 @@ executor type in your `.circleci/config.yml` to be `machine` instead of the defa
 Here is a sample CircleCI configuration that does a checkout of a project and runs maven:
```
Comment (kiview) [inline docs/supported_docker_environment/continuous_integration/circle_ci.md:6]:
```suggestion
Here is a sample CircleCI configuration (applicable for Cloud, Server v2.x, and Server v3.x)  that does a checkout of a project and runs maven:
```

**Review by kiview — APPROVED** [review]
Thanks for the pointer @epragtbeamtree. I took this opportunity to further cleanup this part of the docs a bit 🙂

**Comment by epragtbeamtree** [comment]
@kiview no worries, glad to see it got merged, thanks!

**Pull request merged**